### ISSUE TITLE
Exception using stagemonitor jdbc

### ISSUE THREAD

**Issue by gauravrmazra** [stagemonitor/stagemonitor#136]
Hi,

I am getting following exception running stagemonitor in Spring Boot App.

```
java.util.ServiceConfigurationError: org.stagemonitor.core.instrument.StagemonitorJavassistInstrumenter: Provider org.stagemonitor.jdbc.ConnectionMonitoringInstrumenter could not be instantiated
        at java.util.ServiceLoader.fail(Unknown Source)
        at java.util.ServiceLoader.access$100(Unknown Source)
        at java.util.ServiceLoader$LazyIterator.nextService(Unknown Source)
        at java.util.ServiceLoader$LazyIterator.next(Unknown Source)
        at java.util.ServiceLoader$1.next(Unknown Source)
        at org.stagemonitor.core.instrument.MainStagemonitorClassFileTransformer.<init>(MainStagemonitorClassFileTransformer.java:54)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
        at java.lang.reflect.Constructor.newInstance(Unknown Source)
        at java.lang.Class.newInstance(Unknown Source)
        at org.stagemonitor.agent.StagemonitorAgent.getMainStagemonitorClassFileTransformer(StagemonitorAgent.java:59)
        at org.stagemonitor.agent.StagemonitorAgent.access$100(StagemonitorAgent.java:8)
        at org.stagemonitor.agent.StagemonitorAgent$1.initClassFileTransformer(StagemonitorAgent.java:44)
        at org.stagemonitor.agent.StagemonitorAgent$1.transform(StagemonitorAgent.java:36)
        at sun.instrument.TransformerManager.transform(Unknown Source)
        at sun.instrument.InstrumentationImpl.transform(Unknown Source)
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(Unknown Source)
        at java.security.SecureClassLoader.defineClass(Unknown Source)
        at java.net.URLClassLoader.defineClass(Unknown Source)
        at java.net.URLClassLoader.access$100(Unknown Source)
        at java.net.URLClassLoader$1.run(Unknown Source)
        at java.net.URLClassLoader$1.run(Unknown Source)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(Unknown Source)
        at org.springframework.boot.loader.LaunchedURLClassLoader.doLoadClass(LaunchedURLClassLoader.java:158)
        at org.springframework.boot.loader.LaunchedURLClassLoader.loadClass(LaunchedURLClassLoader.java:130)
        at java.lang.ClassLoader.loadClass(Unknown Source)
        at ch.qos.logback.classic.Logger.buildLoggingEventAndAppend(Logger.java:440)
        at ch.qos.logback.classic.Logger.filterAndLog_0_Or3Plus(Logger.java:396)
        at ch.qos.logback.classic.Logger.info(Logger.java:600)
        at org.stagemonitor.core.Stagemonitor.doStartMonitoring(Stagemonitor.java:91)
        at org.stagemonitor.core.Stagemonitor.access$000(Stagemonitor.java:23)
        at org.stagemonitor.core.Stagemonitor$1.run(Stagemonitor.java:73)
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.ConcurrentModificationException
        at java.util.Hashtable$Enumerator.next(Unknown Source)
        at com.p6spy.engine.spy.option.SystemProperties.getOptions(SystemProperties.java:36)
        at org.stagemonitor.jdbc.ConnectionMonitor.getP6SpyOptions(ConnectionMonitor.java:91)
        at org.stagemonitor.jdbc.ConnectionMonitor.<init>(ConnectionMonitor.java:48)
        at org.stagemonitor.jdbc.ConnectionMonitoringInstrumenter.<init>(ConnectionMonitoringInstrumenter.java:52)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
        at java.lang.reflect.Constructor.newInstance(Unknown Source)
        at java.lang.Class.newInstance(Unknown Source)
        ... 38 more
```

**Comment by felixbarny** [comment]
Thx for reporting. Does this happen each time or just sometimes. Which version of stagemonitor, which db driver and connection pool are using?

**Comment by felixbarny** [comment]
This seems to be an issue that is caused by p6spy. I've created a pull request for that: https://github.com/p6spy/p6spy/pull/303

**Comment by felixbarny** [comment]
Fixed in 68f8c02. Releasing 0.21.0 now.
